### PR TITLE
Feat/#3 new walk test

### PR DESCRIPTION
**Pull request by FelixFonteneau**
# Pull Request

### Description

Provide code to go through the graph of content and test the broken links.
- Functions to create and register a Virtual User(VU) in alex-server. 
- Functions to request screens to alex-server
- Functions to iterate through all the screens and do:
   + Request new screens if needed
   + See if the screen has links
   + See if the screen contains nil links
   + See if the screen link it self.

I provided some unit tests for the algorithm.

### Closes

#3

### CODE REVIEW

File: test/content_test_platform/alex_serv_tests/broken_link_discovery_test.cljc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                                                                                        :link {:broken "yes"}
+                                                                                        :text "next"}]}]}]
+                                     :type          "SCREEN"}
+                             "id2"  {:id            "id2"
+                                     :link          {:to "id2"}
+                                     :nestedContent [{:id   "test-link"
+                                                      :type "button"
+                                                      :text "next"}
+                                                     {:nestedContent [{:type "button"
+                                                                       :link {:broken "yes"}
```
Comment (fongie) [inline test/content_test_platform/alex_serv_tests/broken_link_discovery_test.cljc:29]: why are these links "broken: yes" instead of "null"?

**Review by fongie — APPROVED** [review]
good job including tests as well! 
an idea for a new issue: it would be nice to be able to run the platform in debug mode and log how the VU travels (visiting CONTENT_ID, found link to XXX, visiting YYY etc). this will make it possible to see at a glance what actually happens
doesnt have to be included in this pr though

File: test/content_test_platform/alex_serv_tests/broken_link_discovery_test.cljc (hunk 1 above)
Comment (FelixFonteneau) [reply test/content_test_platform/alex_serv_tests/broken_link_discovery_test.cljc:29]: Ah, just for visibility of the broken one as the `:to` key will be nil for this link.

Comment (FelixFonteneau) [reply test/content_test_platform/alex_serv_tests/broken_link_discovery_test.cljc:29]: Thanks for this feedback, it is very interesting indeed to do that. I thought also to export a json file as graphviz but seems really less important.

**Pull request merged**